contentmgr/replication.go

```diff
@@ -297,17 +234,33 @@ func (cm *ContentManager) ToCheck(contID uint) {
 	}
 }
 
+func (cm *ContentManager) getReadyStagingZones() ([]util.Content, error) {
+	var readyZones []util.Content
+	if err := cm.DB.Model(&util.Content{}).
+		Where("not active and pinning and aggregate and size >= ?", constants.MinDealContentSize).
+		Find(&readyZones).Error; err != nil {
+		return nil, err
+	}
+	return readyZones, nil
+}
+
 func (cm *ContentManager) runStagingBucketWorker(ctx context.Context) {
 	timer := time.NewTicker(cm.cfg.StagingBucket.AggregateInterval)
 	for {
 		select {
 		case <-timer.C:
 			cm.log.Debugw("content check queue", "length", len(cm.queueMgr.queue.elems), "nextEvent", cm.queueMgr.nextEvent)
 
-			zones := cm.popReadyStagingZone()
-			for _, z := range zones {
```

Comment:
@neelvirdy Initially, we pop the zone out so new content is not added when aggregating. Now that we do not pop, we need to track it, or else content will be added to an ongoing aggregation which might lead to an unpredictable outcome. 

I say we change the `ZonesConsolidating` map to something like `processingZones` and use that to track both aggregation and consolidation

Reply:
this was a bit tricky due to it being difficult to tell when consolidating is done if the destination is not the local node. i think there's still room for improvement there, but ideally we remove the concept of consolidation instead. 

"Processing" a ready zone can go through roughly 3 paths:
1. consolidation is done with the local node as destination
2. CMD_TakeContent will be sent to consolidate content to a non-local node
3. consolidation isn't needed and just aggregation will be done

In cases (1) and (2), another "processing" phase will have to happen and (1) or (2) might happen again, if more content was added to the zone. If not, (3) will happen and that will be the last time that zone is "processing" before it is set as active.

I only have a hotspot connection where I'm staying for the next few days so I wasn't able to test this thoroughly yet. But if it's something we can test in staging or any pre-release environment, would be worth digging into it to validate.

Reply:
Here are the scenarios (and why we need to track processing);

- A ready zone with all content in the same location - it will run aggregation. When aggregation is running, it is not safe to add new content and to have a concurrent run. This scenario currently uses the pin complete to tell when it is done.
- A ready zone with contents in different locations - it will run consolidation. When consolidation is running, it is safe to add new content, but not safe to have a concurrent run. The scenario can use the worker re-run to check the completion by checking the location of the content and remove the flag if all are in the same location (consolidated), then proceed to aggregation

In either case, we need to mark an ongoing run and react accordingly (prevent re-run and adding new content for aggregates, prevent re-run for consolidation).

Reply:
should be fixed - pending testing with shuttles